xarray/core/missing.py

```diff
@@ -0,0 +1,338 @@
+        elif fill_value is None:
+            fill_value = np.nan
+
+        self.f = interp1d(xi, yi, kind=self.kind, fill_value=fill_value,
+                          bounds_error=False, **self.cons_kwargs)
+
+
+class FromDerivativesInterpolator(BaseInterpolator):
+    '''Piecewise polynomial in terms of coefficients and breakpoints.
+
+    See Also
+    --------
+    scipy.interpolate.BPoly
+    '''
+    def __init__(self, xi, yi, kind=None, fill_value=None, **kwargs):
+        from scipy.interpolate import BPoly
+
+        if kind is None:
+            raise ValueError('kind is a required argument')
+
+        self.kind = kind
+        self.cons_kwargs = kwargs
+
+        if fill_value is not None:
+            raise ValueError('from_derivatives does not support fill_value')
+
+        self.f = BPoly.from_derivatives(xi, yi, **self.cons_kwargs)
+
+
+class SplineInterpolator(BaseInterpolator):
+    '''One-dimensional smoothing spline fit to a given set of data points.
+
+    See Also
+    --------
+    scipy.interpolate.UnivariateSpline
+    '''
+    def __init__(self, xi, yi, kind=None, fill_value=None, order=3, **kwargs):
+        from scipy.interpolate import UnivariateSpline
+
+        if kind is None:
+            raise ValueError('kind is a required argument')
+
+        self.kind = kind
+        self.cons_kwargs = kwargs
+        self.call_kwargs['nu'] = kwargs.pop('nu', 0)
+        self.call_kwargs['ext'] = kwargs.pop('ext', None)
+
+        if fill_value is not None:
+            raise ValueError('SplineInterpolator does not support fill_value')
+
+        self.f = UnivariateSpline(xi, yi, k=order, **self.cons_kwargs)
+
+
+def get_clean_interp_index(arr, dim, use_coordinate=True, **kwargs):
+    '''get index to use for x values in interpolation.
+
+    If use_coordinate is True, the coordinate that shares the name of the
+    dimension along which interpolation is being performed will be used as the
+    x values.
+
+    If use_coordinate is False, the x values are set as an equally spaced
+    sequence.
+    '''
+
+    if use_coordinate:
+        index = arr.get_index(dim)
+        if isinstance(index, pd.DatetimeIndex):
+            index = index.values.astype(np.float64)
+
+        # check index sorting now so we can skip it later
+        if not (np.diff(index) > 0).all():
+            raise ValueError("Index must be monotonicly increasing")
+
+    else:
+        axis = arr.get_axis_num(dim)
+        index = np.arange(arr.shape[axis])
+
+    return index
+
+
+def interp_na(self, dim=None, use_coordinate=True, method='linear', limit=None,
+              inplace=False, **kwargs):
+    '''Interpolate values according to different methods.'''
+
+    # this may not be possible with apply_ufunc
+    arr = self if inplace else self.copy()
+
+    if dim is None:
+        raise NotImplementedError('dim is a required argument')
+
+    if limit is not None:
+        valids = _get_valid_fill_mask(arr, dim, limit)
```

Comment:
@dcherian - the limit option is now supported.

Reply:
Awesome. Thanks!